tests/paddle/test_paddle_autolog.py

```diff
@@ -0,0 +1,194 @@
+
+    model.fit(train_dataset, eval_dataset, epochs=NUM_EPOCHS, batch_size=8, verbose=1)
+
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(client.list_run_infos(experiment_id="0")[0].run_id)
+    return model, run
+
+
+@pytest.mark.parametrize("log_models", [True, False])
+def test_paddle_autolog_log_models_configuration(log_models):
+    mlflow.paddle.autolog(log_models=log_models)
+
+    paddle.set_default_dtype("float32")
+    train_dataset = paddle.text.datasets.UCIHousing(mode="train")
+    eval_dataset = paddle.text.datasets.UCIHousing(mode="test")
+
+    class UCIHousing(paddle.nn.Layer):
+        def __init__(self):
+            super(UCIHousing, self).__init__()
+            self.fc = paddle.nn.Linear(13, 1)
+
+        def forward(self, feature):  # pylint: disable=arguments-differ
+            pred = self.fc(feature)
+            return pred
+
+    model = paddle.Model(UCIHousing())
+    optim = paddle.optimizer.Adam(learning_rate=0.01, parameters=model.parameters())
+    model.prepare(optim, paddle.nn.MSELoss())
+
+    model.fit(train_dataset, eval_dataset, epochs=NUM_EPOCHS, batch_size=8, verbose=1)
+
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(client.list_run_infos(experiment_id="0")[0].run_id)
+    run_id = run.info.run_id
+    client = mlflow.tracking.MlflowClient()
+    artifacts = [f.path for f in client.list_artifacts(run_id)]
+    assert ("model" in artifacts) == log_models
+
+
+def test_paddle_autolog_logs_default_params(paddle_model):
+    _, run = paddle_model
+    data = run.data
+    assert "optimizer_name" in data.params
+
+
+def test_paddle_autolog_logs_expected_data(paddle_model):
+    _, run = paddle_model
+    data = run.data
+
+    # Checking if metrics are logged
+    client = mlflow.tracking.MlflowClient()
+    for metric_key in ["batch_size", "loss", "step", "eval_batch_size", "eval_loss", "eval_step"]:
+        assert metric_key in run.data.metrics
+        metric_history = client.get_metric_history(run.info.run_id, metric_key)
+        assert len(metric_history) == NUM_EPOCHS
+
+    # Testing optimizer parameters are logged
+    assert "optimizer_name" in data.params
+    assert data.params["optimizer_name"] == "Adam"
+
+    # Testing learning rate are logged
+    assert "learning_rate" in data.params
+    assert float(data.params["learning_rate"]) == 1e-2
+
+    # Testing model_summary.txt is saved
+    client = mlflow.tracking.MlflowClient()
+    artifacts = client.list_artifacts(run.info.run_id)
+    artifacts = map(lambda x: x.path, artifacts)
+    assert "model_summary.txt" in artifacts
+
+
+def test_paddle_autolog_persists_manually_created_run():
+    with mlflow.start_run() as manual_run:
+        mlflow.paddle.autolog()
+
+        paddle.set_default_dtype("float32")
+        train_dataset = paddle.text.datasets.UCIHousing(mode="train")
+        eval_dataset = paddle.text.datasets.UCIHousing(mode="test")
+
+        class UCIHousing(paddle.nn.Layer):
```

Comment:
Can we move this class in the top level?

Reply:
Done